src/client.rs

```diff
@@ -91,7 +100,7 @@ where
     /// Returns `Ready` when the connection can initialize a new HTTP 2.0
     /// stream.
     pub fn poll_ready(&mut self) -> Poll<(), ::Error> {
-        self.connection.poll_send_request_ready()
+        self.inner.poll_pending_open(self.pending.as_ref())
```

Comment:
We may want to add a `fn ready(self) -> impl Future` to ease dealing with this.

I also think it may be nice to add a similar future to `Stream` for when capacity is available (maybe combines with when the pending stream is opened, and so a user could send data, depends on dicussion lower in this PR).